src/main/java/com/liefery/android/stop_badge/StopBadgeFactory.java

```diff
@@ -0,0 +1,38 @@
+package com.liefery.android.stop_badge;
+
+import android.annotation.SuppressLint;
+import android.graphics.*;
+import android.util.Log;
+import android.util.LruCache;
+
+@SuppressLint( "NewApi" )
+public class StopBadgeFactory {
+
+    private static StopBadgeFactory instance = null;
+
+    private BadgeCache cache = new BadgeCache();
+
+    protected StopBadgeFactory() {
+        // Exists only to defeat instantiation.
+    }
+
+    public static Bitmap getBitmap( StopBadge badge, int size ) {
+        if ( instance == null ) {
+            instance = new StopBadgeFactory();
+        }
+
+        if ( instance.cache.get( badge.toKey( size ) ) == null ) {
+            Bitmap newBadge = badge.export( size );
+            instance.cache.put( badge.toKey( size ), newBadge );
+            return newBadge;
+        } else {
+            return instance.cache.get( badge.toKey( size ) );
+        }
+    }
+
+    public class BadgeCache extends LruCache<String, Bitmap> {
+        BadgeCache() {
+            super( 2 * 1024 * 1024 ); // 2MiB
```

Comment:
2 MB cache is more than sufficient. On my device 20dp (the size of the badge is 131px). 

```
2097152                         /   52400                           ≈     40
2mb cache in bytes           bytes per 131px bmp           StopBadges per cache

Above calculation applies to a OnePlus 5, values might differ on other devices.
```